jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -56,38 +51,98 @@
 {
     private static final Logger LOGGER = Log.getLogger(Pool.class);
 
-    private final List<Entry> sharedList = new CopyOnWriteArrayList<>();
+    private final List<Entry> entries = new CopyOnWriteArrayList<Entry>()
+    {
+        @Override
+        public ListIterator<Entry> listIterator(int index)
+        {
+            try
+            {
+                return super.listIterator(index);
+            }
+            catch (IndexOutOfBoundsException e)
+            {
+                return listIterator();
+            }
+        }
+    };
+
+    private final int maxEntries;
+    private final AtomicInteger pending = new AtomicInteger();
+    private final StrategyType strategyType;
+
     /*
      * The cache is used to avoid hammering on the first index of the entry list.
      * Caches can become poisoned (i.e.: containing entries that are in use) when
      * the release isn't done by the acquiring thread or when the entry pool is
      * undersized compared to the load applied on it.
      * When an entry can't be found in the cache, the global list is iterated
-     * normally so the cache has no visible effect besides performance.
+     * with the normal strategy so the cache has no visible effect besides performance.
```

Comment:
There is no "normal" strategy, I would rephrase: "is iterated with the configured StrategyType".